test-case/check-keyword-detection-in-D0i3.sh

```diff
@@ -0,0 +1,164 @@
+
+# get wov PGA volume kcontrol name
+wov_kctl_pga=$(amixer controls |grep "KWD Capture Volume" |awk -F= '{print $4}')
+
+# adjust wov PGA volume
+amixer cset name="$wov_kctl_pga" 40
+
+# store the default config blob
+_store_default_wov_config_bolb()
+{
+    if [ ! -f "$def_blob" ]; then
+        sof-ctl -Dhw:0 -n "$wov_kctl_id" -r -o "$def_blob"
+        dlogi "default blob:"
+        cat "$def_blob"
+    fi
+}
+
+# get default preamble time and history depth
+_get_default_pt_hd()
+{
+    def_pt=$(awk -F "," '{print $4}' < "$def_blob")
+    def_hd=$(awk -F "," '{print $7}' < "$def_blob")
+}
+# restore the default config blob
+_restore_default_wov_config_bolb()
+{
+    if [ -f "$def_blob" ]; then
+        sof-ctl -Dhw:0 -n "$wov_kctl_id" -r -s "$def_blob"
+    fi
+}
+
+# update the blob
+_update_blob(){
+    new_blob=$test_dir"/new_blob"
+    [ "$preamble_time" -ge "$history_depth" ] || {
+        die "Warning: invalid arguments, preamble_time must be greater than or equal to history_depth"
+    }
+    awk -F, -v OFS=, '{if ( $4 == '"$def_pt"' && $7 == '"$def_hd"' ) $4='"$preamble_time"'; $7='"$history_depth"'}1' \
+        "$def_blob" > "$new_blob"
+    dlogi "kwd config blob is updated to:"
+    cat "$new_blob"
+    dlogi "write back the new kwd config blob"
+    sof-ctl -Dhw:0 -n "$wov_kctl_id" -r -s "$new_blob" >> /dev/null || die "Failed to write back the new kwd config bolb"
+}
+
+for i in $(seq 1 "$loop_cnt")
+do
+    for fmt in $fmts
+    do
+        if [ "$fmt" == "S24_LE" ]; then
+            dlogi "S24_LE is not supported, skip to test this format"
+            continue
+        fi
+        test_file="$test_dir"/wov_"${fmt%_*}"_test.wav
+        recorded_file="$test_dir"/wov_"${fmt%_*}"_record.wav
+        _store_default_wov_config_bolb || die "Failed to dump KWD config blob "
+        _get_default_pt_hd || die "Failed to get the default preamble_time and history_depth"
+        _update_blob || die "Failed to update the KWD config blob"
+        # generate the wav files for test
+        wavetool.py -g wov -A 0.25 1. -D 4. 6. -B "${fmt%_*}" -o "$test_file"
+
+        # run arecord on keyword detection pipeline first
+        dlogc "arecord -D$dev -M -N -r $rate -c $channel -f $fmt --buffer-size=$buffer_size -d $duration $recorded_file"
+        arecord -D"$dev" -M -N -r "$rate" -c "$channel" -f "$fmt" --buffer-size="$buffer_size" -d "$duration" \
+		"$recorded_file" -vvv 2>/dev/null & WOVPID=$! 
+        
+	sleep 6 # wait for DSP enter to D0i3
+	# check D0i3 status
+	sudo hexdump /sys/kernel/debug/sof/hda |grep "0001040" |awk -F ' ' '{print $7}' |grep "04" || die "DSP failed to enter D0i3"
+
+        # play the test wav to trigger kwd
+        dlogc "aplay -Dplughw:1,0 $test_file"
+        aplay -Dplughw:1,0 "$test_file"  || die "Failed to play the test wav on USB sound card"
+        dlogi "Testing: iteration $i with $fmt format"
+
+        timeout 5 tail --pid="$WOVPID" -f /dev/null || die "Failed to stop arecord on kwd pipeline"
```

Comment:
We already discussed this pattern a few times:
- Don't use `tail --pid` to wait for a $WOVPID, use `wait $WOVPID`. 
- Use the timeout on the arecord command itself, not on the wait.
- Also use the `--kill-after=` long option, it's safer and more English makes the code more readable.